libs/pilight/protocols/433.92/teletaster.c

```diff
@@ -0,0 +1,221 @@
+/*
+	Copyright (C) 2016 ramyres
+
+	This file is part of pilight.
+
+	pilight is free software: you can redistribute it and/or modify it under the
+	terms of the GNU General Public License as published by the Free Software
+	Foundation, either version 3 of the License, or (at your option) any later
+	version.
+
+	pilight is distributed in the hope that it will be useful, but WITHOUT ANY
+	WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+	A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with pilight. If not, see	<http://www.gnu.org/licenses/>
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+#include "../../core/pilight.h"
+#include "../../core/common.h"
+#include "../../core/dso.h"
+#include "../../core/log.h"
+#include "../protocol.h"
+#include "../../core/binary.h"
+#include "../../core/gc.h"
+#include "teletaster.h"
+
+#define PULSE_MULTIPLIER	4
+#define MIN_PULSE_LENGTH	450
+#define MAX_PULSE_LENGTH	480
+#define AVG_PULSE_LENGTH	465
+#define RAW_LENGTH				36
+
+#define ID_LENGTH 9
+
+static int validate(void) {
+	if(teletaster->rawlen == RAW_LENGTH) {
+		if(teletaster->raw[teletaster->rawlen-1] >= (MIN_PULSE_LENGTH*PULSE_DIV) &&
+		   teletaster->raw[teletaster->rawlen-1] <= (MAX_PULSE_LENGTH*PULSE_DIV)) {
+			return 0;
+		}
+	}
+
+	return -1;
+}
+
+static void createMessage(char* id) {
+	teletaster->message = json_mkobject();
+	json_append_member(teletaster->message, "id", json_mkstring(id));}
+
+static void parseCode(void) {

```

Comment:
Where is the on/off decoding?

Reply:
There ise non. May I refer you again to https://forum.pilight.org/Thread-Teletaster-protocol

Reply:
Does a `contact` not fit better?

Reply:
When I implemented this last year, I was scanning through the existing protocols and had the impression that `contact` is a "passive" element that only reports its state, but cannot be actively triggered to change it. This Teletaster is like a push button: You send the signal and toggle the state, but you can't derive the current state of the switch.